Add setup instructions for development in README.md (#2)
falk-werner/note.rs: README.md

Yet another note taking tool.

## Build and run
This project is build with [rust](https://www.rust-lang.org/) and [npm](https://www.npmjs.com/).
Please see to have it installed beforehand.

Additionally, running `note.rs` will create a __GUI__ and it needs a display for this.
See to have a setup which enables this.

### Install dependencies
`note.rs` uses [tauri](https://tauri.app/) and [vite](https://vitejs.dev/).

> <details><summary>How to setup <b>Ubuntu 20.04</b> with tauri and vite.</summary>

Reviewer: Please don't use HTML tags.

Author: To add an collapsible section html is used as described [here](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-collapsed-sections)
Do you wish to renounce on the collapsible for the sake of no html in markdown?

Reviewer: yes

Reviewer: It is nice, that this featue is recognized by github, but I think we don't need it now. We don't have too much text by now.

Author: With commit d4eb70adee4fcbe0b3a848c9d816860ef81a9312 I've removed the html tags.

cargo tauri dev
```

> ## Setup environment

Reviewer: I prefer a rather short "Build and Run" section. See [note.js readme](https://github.com/falk-werner/note.js/tree/feature/update-readme) for reference. We should merge the setup with the previous section, where the build instructions should preceed the run instructions.

Author: With commit c337707716c614b8271430b3999e47c3e33cc656 I tried to merge it together without overloading the screen with too much information.
Especially, I hide the steps for installing the dependencies behind a collapsible section to encourage a reader to look up both tauri and vite.

> This guide focuses on how to setup your development system based on ubuntu:jammy.

Reviewer: Since we produce an appimage and appimage build is bound to the oldest supported platform*, ubuntu:20.04 (focal) should be the reference platform. I also prefer the actual version number instead of the name. Please change.

```suggestion
This guide focuses on how to setup your development system based on ubuntu:20.04.
```

_(*this is due to libc dependencies, see [Binaries compiled on old enough base system](https://github.com/AppImage/AppImageKit/wiki/Creating-AppImages/cc2441518975caca23e9ce2dba6f08a22c678d1e))_

Author: With commit c337707716c614b8271430b3999e47c3e33cc656 the version number is used and the information of the use of ubuntu is now moved to the summary of the collapsible section. Is the use of Ubuntu 20.04 a __must__ or is it possible to build and run the project on other OS? I think by referencing Ubuntu 20.04 in the collapsible section header relatives Ubuntu 20.04  as the OS for development.

Reviewer: It is not a must, but recommended due to appimage requirements. When you don't need to support Ubuntu 20.04, you can use a newer version and / or a different distribution.

1. Install dependencies
    ```bash
    sudo apt-get update
    sudo apt-get install -y \
        libwebkit2gtk-4.0-dev \

>         build-essential \
>         curl \
>         wget \
>         file \

Reviewer: We do not depend on these packages (see [build workflow](https://github.com/falk-werner/note.rs/blob/main/.github/workflows/build.yaml)), please remove it.
```suggestion
```

Author: This packages are removed with commit c337707716c614b8271430b3999e47c3e33cc656.

>         libssl-dev \

Reviewer: The same goes here: we do not directly depend on the package, please remove it.

```suggestion
```

Author: This package is remove with commit c337707716c614b8271430b3999e47c3e33cc656.

        libgtk-3-dev \
        libayatana-appindicator3-dev \
        librsvg2-dev
    ```

> 2. Install rust
>     ```bash
>     curl --proto '=https' --tlsv1.2 https://sh.rustup.rs -sSf | sh
>     ```

Reviewer: This section should be removed, just state that rust is needed. There are multiple ways to install rust depending on your personal preferences and requirements. Do not force a user to a specific way.

```suggestion
```

Author: This section is removed with commit c337707716c614b8271430b3999e47c3e33cc656 and only states the need to have it installed beforehand.

> 3. Install npm (node.js)
>     ```bash
>     curl -fsSL https://deb.nodesource.com/setup_20.x | sudo -E bash - &&\
>     sudo apt-get install -y nodejs
>     ```

Reviewer: Some goes here. Please remove the section and just state, that npm is needed. For npm there are just more installation methods than for rust.

```suggestion
```

Author: This section is removed with commit c337707716c614b8271430b3999e47c3e33cc656 and only states the need to have it installed beforehand.
